### ISSUE TITLE
Media elements that delay the load event...

### ISSUE THREAD

**Issue by travisleithead** [w3c/html#428]
MSE's MediaSource will likely not delay the load event as soon as it is attached to the media element. Same should apply for a MediaStream. Ensure the spec is clear on this point.

**Comment by foolip** [comment]
AFAICT, Blink doesn't have any extra "Set the element's delaying-the-load-event flag to false" beyond what the spec already requires. Given that the flag is set to true in the resource selection algorithm before one knows what kind of resource will be used, at what point would it be set to false? Does any engine already do this?

**Comment by travisleithead** [comment]
My read: The [resource selection algorithm](http://w3c.github.io/html/semantics-embedded-content.html#resource-selection-algorithm), step 3 sets the flag to true, this can happen while the parser is active on the stack and the load event hasn't fired yet. For a `MediaSource` assignment, aside from an error, the [resource fetch algorithm](http://w3c.github.io/html/semantics-embedded-content.html#resource-fetch-algorithm) invokes the [media data processing steps list](http://w3c.github.io/html/semantics-embedded-content.html#media-data-processing-steps-list) which starts examining the media content, and when those steps eventually successfully complete, then `readyState` goes to `HAVE_CURRENT_DATA`, the `loadeddata` event is fired, and finally the delay-the-load-event flag goes to false.

There are some async steps in there, so it's not "immediate", but in layman's terms, as soon as the media source data can be made available (first data available), then the element stops delaying the load event. I think that's close to the answer we arrived at with the folks on the MSE teleconference. In implementation, they may just set the flag to false on assignment of the `srcObject`. Given that author-provided media could be arbitrarily delayed (even before initial media data can be fed?), I like putting in an explicit step to set the flag to false, possibly earlier than it is currently described in the spec.

...need to find a link to their issue.

**Comment by foolip** [comment]
Right, "Once the`readyState` attribute reaches `HAVE_CURRENT_DATA`, after the `loadeddata` event has been fired, set the element's delaying-the-load-event flag to false. This stops delaying the load event."

There is nothing else that would set the flag to false earlier for MSE, so the question is when to do it, and if some engine already does? I think that in the resource fetch algorithm is where it would make sense. That's also where preload=none is handled, and with https://github.com/whatwg/html/pull/1037 fixed it could all be in the "Otherwise (mode is local)" branch.

**Comment by paulbrucecotton** [comment]
@travisleithead - Will this issue be resolved in HTML 5.1?

**Comment by chaals** [comment]
@travisleithead can you get this done in time for the next draft? Otherwise please shift the milestone.

**Comment by travisleithead** [comment]
Paul followed up offline with me and reported that there is no need to get this done in 5.1.  MSE solved this problem on its own.

Updated the milestone.
